hibernate-types-52/src/main/java/com/vladmihalcea/hibernate/type/money/MonetaryAmountType.java

```diff
@@ -75,14 +75,14 @@ public int hashCode(Object x) throws HibernateException {
 
     @Override
     public Object nullSafeGet(ResultSet rs, String[] names, SharedSessionContractImplementor session, Object owner) throws HibernateException, SQLException {
-        if (rs.wasNull()) {
-            return null;
-        }
-
         String amountColumnName = names[0];
         String currencyColumnName = names[1];
 
         BigDecimal amount = rs.getBigDecimal(amountColumnName);
+        if(amount == null) {
```

Comment:
Removing this null-check in favor of `rs.wasNull()` (see next comment).

Reply:
@piotrooo The `wasNull` method call is useful only when you have primitive values on the Java side, but this is not the case here. We can use `== null` checks.

Reply:
`null` seems to be enough.